### ISSUE TITLE
Sile prints vUNKNOWN when running

### ISSUE THREAD

**Issue by clintolsen** [sile-typesetter/sile#807]
I'm not very adept with brew, but when I run sile I see:

> This is SILE vUNKNOWN

**Comment by alerque** [comment]
Thank you for the report @clintolsen, but I may have fixed this already in e46091f7. What version exactly did you install? This looks like something that would turn up in the `--HEAD` builds, but it should also be fixed in the very latest HEAD so you would have to have built before the 29th to turn this up.

I don't know how we can prove this until after the next release cycle because building it manually to test doesn't show up the same problem as the actual build bot does. There will probably be a another point release very soon with some fixes needed for Ubuntu builds.

**Comment by alerque** [comment]
Sorry @clintolsen just to clarify my question, did you:

a) install with `brew install sile` or
b) install with `brew install sile --HEAD`

And secondarily, which day did you install on? That should give me an idea whether this should be fixed since your install or whether something is still broken.

**Comment by clintolsen** [comment]
It appears this was mostly operator error. Brew is an obtuse system. It's
not clear what's going on here:

poly ~/Documents/sile$ sile -v
SILE vUNKNOWN

poly ~/Documents/sile$ brew install sile
Error: sile HEAD-e46091f is already installed
To install 0.10.1, first run `brew unlink sile`.
poly ~/Documents/sile$ brew unlink sile
Unlinking /usr/local/Cellar/sile/HEAD-e46091f... 8 symlinks removed
poly ~/Documents/sile$ brew upgrade sile
Warning: sile HEAD-e46091f already installed
poly ~/Documents/sile$ sile -v
-bash: /usr/local/bin/sile: No such file or directory
poly ~/Documents/sile$ brew install sile
==> Downloading
https://homebrew.bintray.com/bottles/sile-0.10.1.mojave.bottle.tar.gz
==> Downloading from
https://akamai.bintray.com/ca/ca6e60229ac5f6a6a9e0554c5fb1d2aecc3807556ae7b57943602cfc05061b20?__gd
########################################################################
100.0%
==> Pouring sile-0.10.1.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/sile/0.10.1: 792 files, 11.0MB
poly ~/Documents/sile$ sile -v
SILE v0.10.1

At any rate, it's working OK now.

Thanks,

-Clint

**Comment by alerque** [comment]
Thanks for the extra info @clintolsen. I'm glad it's working for you now, but unfortunately the evidence is that the problem isn't fixed yet. You previously had a HEAD version installed and updated that. Unfortunately the e46091f commit should have had the version issue fixed, but it clearly did not for Homebrew. Now you've switched to the stable release version. I'm glad that one works properly and it definitely the more important part of the equation. But we still need to figure out why it didn't work on HEAD. You can safely ignore this issue for now if you like, we'll try to work it out.

@simoncozens are you able to replicate this with a `brew install sile --HEAD`? What `sile --version` do you get?

**Comment by simoncozens** [comment]
I get vUNKNOWN from `brew install sile --HEAD`. Will take a look.

**Comment by simoncozens** [comment]
Homebrew is pulling down a shallow commit with one commit of history, which breaks our version-generator.

```
simon@upo ~/hacks/sile .$ git describe HEAD
v0.10.1-26-g11490500
simon@upo ~/Library/Caches/Homebrew/sile--git *$ git describe --tags HEAD
fatal: No names found, cannot describe anything.
simon@upo ~/Library/Caches/Homebrew/sile--git *$ git log | cat
commit d173db73e1b7e6c2beb9ae33787caff64769c075
Author: Caleb Maclennan <caleb@alerque.com>
Date:   Mon Feb 3 18:37:33 2020 +0300

    Merge pull request #804 from alerque/ubuntu-ppa

    Setup Ubuntu packaging
```

**Comment by alerque** [comment]
Ya we'll need to tell the formula to add depth/tags I think.

**Comment by alerque** [comment]
This should be double fixed, first #818 alleviates the information dearth in sparse checkouts. Additionally when v0.10.3 [hits Homebrew](https://github.com/Homebrew/homebrew-core/pull/49744) it should start using full depth checkouts for HEAD builds, which will give an even more useful version string.
